addOns/jwt/src/main/java/org/zaproxy/zap/extension/jwt/JWTActiveScanner.java

```diff
@@ -0,0 +1,249 @@
+
+        boolean result = new ServerSideAttack(jwtTokenBean, this, param, msg, value).execute();
+        return result;
+    }
+
+    /**
+     * TODO Not sure how can this be implemented. Waits some time to check if token is expired and
+     * then execute the attack. TODO need to implement it.
+     *
+     * @param jwtTokenBean
+     * @param msg
+     * @param param
+     * @return
+     */
+    private boolean checkExpiredTokenAttack(
+            JWTTokenBean jwtTokenBean, HttpMessage msg, String param) {
+        return false;
+    }
+
+    public void raiseAlert(
+            int risk,
+            int confidence,
+            String name,
+            String description,
+            String uri,
+            String param,
+            String attack,
+            String otherInfo,
+            String solution,
+            HttpMessage msg) {
+        newAlert()
+                .setRisk(risk)
+                .setConfidence(confidence)
+                .setName(name)
+                .setDescription(description)
+                .setUri(uri)
+                .setParam(param)
+                .setAttack(attack)
+                .setOtherInfo(otherInfo)
+                .setSolution(solution)
+                .setMessage(msg)
+                .raise();
+    }
+
+    /**
+     * @param msg
+     * @param param
+     * @param jwtToken
+     * @return {@code true} if the vulnerability is found, {@code false} otherwise.
+     */
+    public boolean sendManipulatedMsgAndCheckIfAttackSuccessful(
+            HttpMessage msg, String param, String jwtToken, String value) {
+        HttpMessage newMsg = this.getNewMsg();
+        this.setParameter(newMsg, param, JWTUtils.addingJWTToParamValue(value, jwtToken));
+        try {
+            this.sendAndReceive(newMsg, false);
+            if (newMsg.getResponseHeader().getStatusCode()
+                    == msg.getResponseHeader().getStatusCode()) {
+                return true;
+            }
+        } catch (IOException e) {
+            LOGGER.error("Following exception occurred while sending manipulated jwt message", e);
+        }
+        return false;
+    }
+
+    @Override
+    public int getId() {
+        return PLUGIN_ID;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public String getDescription() {
+        return DESCRIPTION;
+    }
+
+    @Override
+    public String getSolution() {
+        return "https://cheatsheetseries.owasp.org/cheatsheets/JSON_Web_Token_Cheat_Sheet_for_Java.html";
+    }
+
+    @Override
+    public String getReference() {
+        return "https://tools.ietf.org/html/draft-ietf-oauth-jwt-bcp-06";
+    }
+
+    @Override
+    public TechSet getTechSet() {
+        TechSet techSet = super.getTechSet();
+        if (techSet != null) {
+            return techSet;
+        }
+        return TechSet.AllTech;
+    }
+
+    @Override
+    public int getCategory() {
+        return 0;
```

Comment:
not sure, it can be a info gather also as we are finding issues with the browser side implementation of jwt and telling about issues and also we are injecting some of the different tokens to find how application behaves so thinking of adding under Misc, @kingthorin what do you think.

Reply:
I’m good with Misc.